src/Store.ts

```diff
@@ -533,6 +533,7 @@ const stopListening: Epic<ChatActions, ChatState> = (action$) =>
 
 const startListening:Epic<ChatActions, ChatState> = (action$, store) =>
     action$.ofType('Listening_Starting')
+    .throttleTime(500)
```

Comment:
can you explain this? Where did the 500ms number come from? Is this working around a latency problem, and if so what happens if that thing we're working around takes 501ms?

Reply:
The issue is that the following operations on the recognizer aren't supported in quick succession:
Start, Stop, Start

Stop should ideally be sync (or async with completion) but isn't. The mitigation is to throttle the signal to start listening (if the user is spamming the button).
The 500ms value is derived empirically based on a mid-ranged laptop (with buffer built in).

Since this is only a mitigation, the worst case scenario theoretically remains the same as today.

Reply:
Look like we need to update [`SpeechModule.ts`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/src/SpeechModule.ts#L104) and making the stop more definitive by signaling when speech is fully stopped, either:
1. Turning `stop()` into a Promise, or,
2. A new event handler

Reply:
@shahidkhuram @compulim what's the consensus on this?

Reply:
Unfortunately it's unlikely that i'll be able to look at this again any time soon. @compulim can you drive any further changes necessary/alternate fixes? If not i suggest taking the change as is at least as a mitigation.

Reply:
@w-Yuan is there any test you can add to simulate this?